### PR TITLE
Content Type Manager: Support multilingual Names/Descriptions in one step

### PR DESCRIPTION
**Pull request by blankse**
Allow the set of all translations for content types and content type fields

e. g.

```
-
    type: content_type
    mode: create
    content_type_group: Content
    identifier: article
    names:
        eng-GB: Article
        ger-DE: Artikel
    descriptions:
        eng-GB: 'Article content type'
        ger-DE: 'Artikel Inhaltsart'
    name_pattern: '<short_title|title>'
    url_name_pattern: ''
    is_container: true
    lang: eng-GB
    attributes:
        -
            identifier: title
            type: ezstring
            names:
                eng-GB: Title
                ger-DE: Titel
            descriptions:
                eng-GB: 'Header of the article'
                ger-DE: 'Überschrift des Artikels'
            required: true
            searchable: true
            info-collector: false
            disable-translation: false
            category: ''
            position: 1
            default-value: 'New article'
            field-settings: {  }
            validator-configuration:
                StringLengthValidator: { maxStringLength: 255, minStringLength: null }

```

### CODE REVIEW

File: Core/Executor/ContentTypeManager.php [hunk 1]
```diff
@@ -64,11 +68,17 @@ protected function create($step)
         $contentTypeCreateStruct->nameSchema = $step->dsl['name_pattern'];
 
         // set names for the content type
-        $contentTypeCreateStruct->names = array(
-            $this->getLanguageCode($step) => $step->dsl['name'],
-        );
+        if (isset($step->dsl['names'])) {
+            $contentTypeCreateStruct->names = $step->dsl['names'];
+        } else {
```
Comment (phibo23) [inline Core/Executor/ContentTypeManager.php]: you forgot to add this new if else for the update function (same for description) ;)

Comment (blankse) [reply Core/Executor/ContentTypeManager.php]: I added it since 5 minutes :)

Comment (phibo23) [reply Core/Executor/ContentTypeManager.php]: ah great! I was just looking at the code again and wondered how I overlooked this earlier. 😁

File: Core/Executor/ContentTypeManager.php [hunk 2]
```diff
@@ -553,9 +567,15 @@ protected function createFieldDefinition(ContentTypeService $contentTypeService,
...
                     $fieldDefinition->names = array($lang => $value);
                     break;
+                case 'names':
+                    $fieldDefinition->names = $value;
+                    break;
                 case 'description':
                     $fieldDefinition->descriptions = array($lang => $value);
                     break;
+                case 'descriptions':
+                    $fieldDefinition->description = $value;
```
Comment (phibo23) [inline Core/Executor/ContentTypeManager.php]: this should be `descriptions` with an `s`, right?

Comment (blankse) [reply Core/Executor/ContentTypeManager.php]: You are right. I changed it.

**Comment by gggeek** [comment]
what about keeping using the same tags 'name' and 'description' and, instead of adding their plural version, just be smart in how we handle them ?
- if user puts in a single string, we assume it to be the value in current lang
- if user puts in an array, then keys are langs
- pretty much nothing changes on create
- on update we merge the user-specified names and descriptions w. pre-existing ones

does it make sense / cover your needs?

**Comment by blankse** [comment]
I changed it to your suggestions 👍

**Pull request merged**